### PR TITLE
Remove magical extra gas computation step right before relaying txn

### PR DESCRIPTION
**Pull request by rweald**
Builds on top of #150 & #149 

----

We were doing some gas estimate mutation in a final step before relaying using padded values returned by the GSN relayer. This padding is okay for dealing with super inconsistent gas spikes on chains with relatively cheap gas payment tokens, but doesn't work when dealing with chains like Base that use ETH. Also was just a bit hacky and hard to follow to mutate gas estimation like this.

### CODE REVIEW

**Comment by rweald** [comment]
I've tested this on Base Sepolia and seen it work end to end. Still need to test on Polygon to ensure backwards compatibility.

**Review by DonGambas — APPROVED** [review]
